### DISCUSSION TITLE
Swift names demangler

### DISCUSSION THREAD

**Discussion by Martyx00** [Vector35/binaryninja-api#2129] [Ideas] [answered]
Hello,
Is demangler for swift names somewhere down the pipeline (https://github.com/apple/swift/blob/main/stdlib/public/runtime/Demangle.cpp)?

Thanks!

**Comment by psifertex** [comment, marked as answer]
No, nothing on the current plan. 

There's an old plugin that implements some demangling: https://github.com/watsonkp/binaryninja-ios-tools

Alternatively, if you're running on a Mac: https://gist.github.com/psifertex/6fbc7532f536775194edd26290892ef7#file-swift_demangler-py

**Reply by psifertex** [reply]
One other comment, you might want to also track https://github.com/Vector35/binaryninja-api/issues/467 which when completed would make it much easier to extend the existing demangler system as an open source plugin.
